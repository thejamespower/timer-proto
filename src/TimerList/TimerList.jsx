import React from 'react'
import PropTypes from 'prop-types'
import Countdown from 'react-sexy-countdown'

const TimerList = ({ timers }) => timers.map(timer => <Countdown key={timer.id} date={timer.date} />)

TimerList.propTypes = {
  timers: PropTypes.arrayOf(
    PropTypes.object,
  ),
}

export default TimerList
