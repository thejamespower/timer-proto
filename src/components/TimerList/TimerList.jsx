import React from 'react';
import PropTypes from 'prop-types';

import Timer from '../Timer';
import SuperTimer from '../SuperTimer';

function TimerList(props) {
  const { timers } = props;
  return (
    <div>
      {timers.map(timer => (
        <Timer key={timer.id} timer={timer} />
      ))}
      <SuperTimer />
    </div>
  );
}

TimerList.propTypes = {
  timers: PropTypes.arrayOf(
    PropTypes.shape({
      duration: PropTypes.string.isRequired,
      durationInSeconds: PropTypes.number.isRequired,
      active: PropTypes.bool.isRequired,
    }),
  ).isRequired,
};

export default TimerList;
