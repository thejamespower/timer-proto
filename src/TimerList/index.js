import TimerList from './container'

export default TimerList
