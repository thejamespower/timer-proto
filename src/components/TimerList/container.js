import { connect } from 'react-redux'

import TimerList from './TimerList'

const mapStateToProps = state => ({
  timers: state.timers.timers,
  superTimer: state.timers.superTimer,
})

export default connect(
  mapStateToProps,
  null,
)(TimerList)
