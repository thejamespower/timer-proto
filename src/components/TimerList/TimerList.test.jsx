import React from 'react'
import renderer from 'react-test-renderer'
import TimerList from './TimerList'

jest.mock('../SuperTimer', () => 'SuperTimer')
jest.mock('../Timer', () => 'Timer')

describe('TimerList', () => {
  let props = {
    timers: [],
  }

  describe('render', () => {
    it('renders default state', () => {
      const component = renderer.create(
        <TimerList {...props} />,
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders timers state', () => {
      props = {
        ...props,
        timers: [...props.timers, {
          id: 'test',
          duration: '00:00:01',
          durationInSeconds: 1,
          name: 'test',
          active: false,
          timeToStart: '00:00:01',
          timeToStartInSeconds: 1,
          complete: false,
        }],
      }

      const component = renderer.create(
        <TimerList {...props} />,
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
