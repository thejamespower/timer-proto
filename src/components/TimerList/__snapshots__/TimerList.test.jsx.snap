// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerList render renders default state 1`] = `
<div>
  <SuperTimer />
</div>
`;

exports[`TimerList render renders timers state 1`] = `
<div>
  <Timer
    timer={
      Object {
        "active": false,
        "complete": false,
        "duration": "00:00:01",
        "durationInSeconds": 1,
        "id": "test",
        "name": "test",
        "timeToStart": "00:00:01",
        "timeToStartInSeconds": 1,
      }
    }
  />
  <SuperTimer />
</div>
`;
